src/path_rewriting.rs

```diff
@@ -16,6 +17,25 @@ fn to_uppercase_first(s: &str) -> String {
     c.next().unwrap().to_uppercase().collect::<String>() + c.as_str()
 }
 
+pub fn canonicalize_path<P: AsRef<Path>>(path: P) -> io::Result<PathBuf> {
+    let path = fs::canonicalize(path)?;
+
+    #[cfg(windows)]
+    let path = match {
```

Comment:
Why use `match` here? This also compiles:
```rust
    #[cfg(windows)]
    let path = {
        let spath = path.to_str().unwrap();
        if spath.starts_with(r"\\?\") {
            PathBuf::from(spath[r"\\?\".len()..].to_string())
        } else {
            path
        }
    };
```

Reply:
It compiles because you aren't under windows ;)